src/libponyc/codegen/genexe.c

```diff
@@ -269,9 +268,9 @@ static bool link_exe(compile_t* c, ast_t* program,
 #ifdef PONY_USE_LTO
     "-flto -fuse-linker-plugin "
 #endif
-    "%s %s %s %s -lpthread %s "
-    "-lm",
-    linker, file_exe, arch, mcx16_arg, fuseld, file_o, lib_args, ponyrt, ldl
+    "-fuse-ld=gold %s %s %s -lpthread %s -lm "
```

Comment:
It was being set for FreeBSD in the Makefile in release mode ([here](https://github.com/ponylang/ponyc/blob/master/Makefile#L157)) so I assumed it was available.

On Linux, the option was added in [this patch](https://gcc.gnu.org/ml/gcc-patches/2012-11/msg02389.html) from 2012, which would make it available on GCC 4.8 and above.